### ISSUE TITLE
Problems with EFI system after `bootc install to-existing-root`

### ISSUE THREAD

**Issue by apollo13** [containers/bootc#996]
So I have been playing with bootc and wanted to try it on a vserver. Sadly I couldn't install it "easily" since all the live cds ran out of storage space when pulling the container image. So I partitioned the existing disk and used part of it as storage for `/var/lib/containers/storage`:

```
root@grml ~ # gdisk -l /dev/vda
Number  Start (sector)    End (sector)  Size       Code  Name
   1            2048         1050623   512.0 MiB   EF00  EFI-SYSTEM
   2         1050624       210765823   100.0 GiB   8300  root
   3       210765824       252708863   20.0 GiB    8300  Linux filesystem # used for the storage
```

I installed the system like this:
```
podman run -it -v /target:/target --rm --tmpfs=/tmp --tmpfs=/var/tmp --privileged --pid=host -v /dev:/dev -v /var/lib/containers:/var/lib/containers -v /data:/data --security-opt label=type:unconfined_t registry.gitlab.com/rocketduck/os bootc install to-existing-root /target
```
The `tmpfs` mounts where necessary otherwise bootc would complain about something similar to https://github.com/containers/bootc/issues/95 (in my case it was `/var/tmp` though). The live system in question is a grml (https://grml.org/).

The install finished successfully:
```
Installing image: docker://registry.gitlab.com/rocketduck/os:latest
Digest: sha256:51e7ff2195e0bd399ede84b555bff4e57b3b94e3b1ffa1a257b2a63ece0013e8
Initializing ostree layout
layers already present: 0; layers needed: 72 (821.7 MB)
Fetched layers: 783.64 MiB in 2 minutes (6.59 MiB/s)                                                                                                                                                                                                                             Deploying container image: done (59 seconds)                                                                                                                                                                                                                                 Injected: etc/tmpfiles.d/bootc-root-ssh.conf
Running bootupctl to install bootloader
> bootupctl backend install --write-uuid --update-firmware --auto --device /dev/vda /target
BootCurrent: 0002
Timeout: 3 seconds
BootOrder: 000A,0002,0003,0004,0005,0006,0007,0000,0001,0008
Boot0000* UiApp	FvVol(64074afe-340a-4be6-94ba-91b5b4d0f71e)/FvFile(462caa21-7614-4503-836e-8ab6f4662331)
Boot0001* UEFI Misc Device	VenHw(93e34c7e-b50e-11df-9223-2443dfd72085,00){auto_created_boot_option}
Boot0002* UEFI QEMU QEMU CD-ROM 	PciRoot(0x0)/Pci(0x2,0x4)/Pci(0x0,0x0)/SCSI(0,0){auto_created_boot_option}
Boot0003* UEFI Misc Device 2	PciRoot(0x0)/Pci(0x2,0x6)/Pci(0x0,0x0){auto_created_boot_option}
Boot0004* UEFI PXEv4 (MAC:8A705ACDC799)	PciRoot(0x0)/Pci(0x2,0x5)/Pci(0x0,0x0)/MAC(8a705acdc799,1)/IPv4(0.0.0.0,0,DHCP,0.0.0.0,0.0.0.0,0.0.0.0){auto_created_boot_option}
Boot0005* UEFI PXEv6 (MAC:8A705ACDC799)	PciRoot(0x0)/Pci(0x2,0x5)/Pci(0x0,0x0)/MAC(8a705acdc799,1)/IPv6([::],0,Static,[::],[::],64){auto_created_boot_option}
Boot0006* UEFI HTTPv4 (MAC:8A705ACDC799)	PciRoot(0x0)/Pci(0x2,0x5)/Pci(0x0,0x0)/MAC(8a705acdc799,1)/IPv4(0.0.0.0,0,DHCP,0.0.0.0,0.0.0.0,0.0.0.0)/Uri(){auto_created_boot_option}
Boot0007* UEFI HTTPv6 (MAC:8A705ACDC799)	PciRoot(0x0)/Pci(0x2,0x5)/Pci(0x0,0x0)/MAC(8a705acdc799,1)/IPv6([::],0,Static,[::],[::],64)/Uri(){auto_created_boot_option}
Boot0008* EFI Internal Shell	FvVol(64074afe-340a-4be6-94ba-91b5b4d0f71e)/FvFile(7c04a583-9e3e-4f1c-ad65-e05268d0b4d1)
Boot000A* Fedora
	HD(1,GPT,66b7cab1-ebf9-43e7-a035-14bfcceb8237,0x800,0x100000)/\EFI\fedora\shimaa64.efi
Installed: grub.cfg
Installation complete!
```

Upon reboot I see grub loading the shim and not doing anything else. I don't have any idea which other information to provide, but I'll add what I think could be useful:
```
root@grml ~ # ls -lisa /dev/disk/by-partlabel /dev/disk/by-label /dev/disk/by-uuid 
/dev/disk/by-label:
total 0
246 0 drwxr-xr-x  2 root root 120 Dec 27 14:18  .
223 0 drwxr-xr-x 11 root root 220 Dec 27 14:18  ..
272 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18  EFI-SYSTEM -> ../../vda1
282 0 lrwxrwxrwx  1 root root   9 Dec 27 14:18 'grml-full-arm64\x202024.12' -> ../../sr0
247 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18  podman -> ../../vda3
252 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18  root -> ../../vda2

/dev/disk/by-partlabel:
total 0
232 0 drwxr-xr-x  2 root root 100 Dec 27 14:18  .
223 0 drwxr-xr-x 11 root root 220 Dec 27 14:18  ..
254 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18  EFI-SYSTEM -> ../../vda1
266 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18 'Linux\x20filesystem' -> ../../vda3
233 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18  root -> ../../vda2

/dev/disk/by-uuid:
total 0
230 0 drwxr-xr-x  2 root root 120 Dec 27 14:18 .
223 0 drwxr-xr-x 11 root root 220 Dec 27 14:18 ..
231 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18 9d85530f-92b7-4356-8e62-5717db01a235 -> ../../vda2
283 0 lrwxrwxrwx  1 root root   9 Dec 27 14:18 2024-12-19-13-49-23-00 -> ../../sr0
261 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18 E32A-ADB9 -> ../../vda1
257 0 lrwxrwxrwx  1 root root  10 Dec 27 14:18 b42a43b4-8c19-4a14-8866-4ce01e553a9c -> ../../vda3
```

The loader entries on the boot folder of the root partition look okay as far as I can tell:
```
root@grml /target/boot # cat loader.1/entries/ostree-1.conf 
title Fedora Linux 41.20241226.0 (Forty One) (ostree:0)
version 1
options root=UUID=9d85530f-92b7-4356-8e62-5717db01a235 rw ostree=/ostree/boot.1/default/580504e8e331e296082c25853c92b6e5e69b51d42734b659452b26133d72478a/0
linux /boot/ostree/default-580504e8e331e296082c25853c92b6e5e69b51d42734b659452b26133d72478a/vmlinuz-6.12.6-200.fc41.aarch64
initrd /boot/ostree/default-580504e8e331e296082c25853c92b6e5e69b51d42734b659452b26133d72478a/initramfs-6.12.6-200.fc41.aarch64.img
abootcfg /ostree/deploy/default/deploy/77f561c776f6c6507558cf34cc4e829748d7c7d21625a05fcfeea599bde0bb29.0/usr/lib/ostree-boot/aboot.cfg
fdtdir /boot/ostree/default-580504e8e331e296082c25853c92b6e5e69b51d42734b659452b26133d72478a/dtb
aboot /ostree/deploy/default/deploy/77f561c776f6c6507558cf34cc4e829748d7c7d21625a05fcfeea599bde0bb29.0/usr/lib/ostree-boot/aboot.img
```

Efi looks okay as well:
```
root@grml /efi # ls *
bootupd-state.json

EFI:
BOOT  fedora

grub2:
bootuuid.cfg  grub.cfg
```

Not sure if the bootuuid should point to the efi partition:
```
root@grml /efi # cat grub2/bootuuid.cfg
set BOOT_UUID="E32A-ADB9"
```

I don't know enough about blscfg which (hopefully) gets executed by grub to tell where it is broken.

**Comment by apollo13** [comment]
So I have managed to boot the machine by changing the `BOOT_UUID` in `bootuuid.cfg` to the UUID of the actual root partition and after a reboot executed `configfile (hd0,gpt1)/grub2/grub.cfg` which brought up the menu. Not sure yet why grub2 doesn't manage to execute that directly yet.

Btw while debugging I also noticed that the EFI partition needs to be labeled `EFI-SYSTEM` or `EFI System Partition` and mine was named `EFI system partition` initially which resulted in bootc not being able to run bootupd (https://github.com/coreos/bootupd/blob/04b905b5aa28f27f004fdc4364ce33536971dd5f/src/efi.rs#L39-L41). I didn't find anything in the docs pointing this out.
